services/QuillLMS/client/app/bundles/Teacher/components/assignment_flow/manage_units/unit.tsx

```diff
@@ -0,0 +1,229 @@
+    return(
+      <div className="activity-cell-container">
+        <DataTableChip color="green" label={name} onClick={(e) => handleDataTableChipClick(e, id)} />
+        {completed && <div className="lesson-complete"><p>Lesson Complete</p><img alt="" src={`${EXTRA_SMALL_ICON_BASE_SRC}/check-circle.svg`} /></div>}
+        {started && <a className={sharedButtonClassName} href={href} rel="noopener noreferrer" target="_blank">Mark As Complete</a>}
+      </div>
+    )
+  }
+
+  function handleLaunchLessonWithNoStudents(e, activityId, classroomUnitId) {
+    if (window.confirm("You have no students in this class. Quill Lessons is a collaborative tool for teachers and students to work together. If you'd like to launch this lesson anyway, click OK below. Otherwise, click Cancel.")) {
+      window.location.href = `${process.env.DEFAULT_URL}/teachers/classroom_units/${classroomUnitId}/launch_lesson/${activityId}`;
+    }
+  }
+
+  function renderLaunchLessonCell(lessonActivity) {
+    const { completed, studentCount, started } = lessonActivity
+    const activityId = getActivityId(lessonActivity)
+    const classroomUnitId = getClassroomUnitId(lessonActivity)
+    if (completed) {
+      return <a className={sharedButtonClassName} href={`/teachers/progress_reports/report_from_classroom_unit_and_activity/${classroomUnitId}/a/${activityId}`} rel="noopener noreferrer" target="_blank">View Report</a>;
+    }
+    if (studentCount === 0) {
+      return <a className={sharedButtonClassName} id="launch-lesson" onClick={(e) => handleLaunchLessonWithNoStudents(e, activityId, classroomUnitId)}>{started ? 'Resume Lesson' : 'Launch Lesson'}</a>;
+    }
+    if (started) {
+      return <a className={sharedButtonClassName} href={`${process.env.DEFAULT_URL}/teachers/classroom_units/${classroomUnitId}/launch_lesson/${activityId}`}>Resume Lesson</a>;
+    }
+    return <a className={sharedButtonClassName} href={`${process.env.DEFAULT_URL}/teachers/classroom_units/${classroomUnitId}/launch_lesson/${activityId}`}>Launch Lesson</a>;
+  }
+
+  function renderCustomizeCell(id) {
+    return(
+      <a className="customize-link focus-on-light" href={`/customize/${id}`} rel="noopener noreferrer" target="_blank">
+        <img alt="extra small pencil icon" src={`${EXTRA_SMALL_ICON_BASE_SRC}/pencil.svg`} />
+      </a>
+    )
+  }
+
+  function renderDownloadCell(id) {
+    return(
+      <a className="download-link focus-on-light" href={`/activities/${id}/supporting_info`} rel="noopener noreferrer" target="_blank">
+        <img alt="extra small arrow down icon" src={`${EXTRA_SMALL_ICON_BASE_SRC}/arrow-down.svg`} />
+      </a>
+    )
+  }
+
+  function renderClassroomActivities() {
+    const lessonsIcon = <img alt="lessons icon" src={`${EXTRA_SMALL_ICON_BASE_SRC}/tool-lessons.svg`} />
+    const rows = [];
+    const activities = getClassroomActivities()
+    activities.forEach((lessonActivity) => {
```

Comment:
You could clean this up a little by doing `const rows = getClassroomActivities().map(lessonActivity => {` instead of pushing to the empty `rows` array.

Reply:
so this data actually comes through as a `Map`, so we have to use forEach to iterate through it and then push that relevant data into the array (I added a comment in the code for better context)